src/http/client.cc

```diff
@@ -125,14 +125,16 @@ future<connection::reply_ptr> connection::recv_reply() {
     http_response_parser parser;
     return do_with(std::move(parser), [this] (auto& parser) {
         parser.init();
-        return _read_buf.consume(parser).then([&parser] {
+        return _read_buf.consume(parser).then([this, &parser] {
             if (parser.eof()) {
                 throw std::runtime_error("Invalid response");
             }
 
             auto resp = parser.get_parsed_response();
             sstring length_header = resp->get_header("Content-Length");
             resp->content_length = strtol(length_header.c_str(), nullptr, 10);
+            // HTTP-1.1 and above only, see comment about this field for details
+            _persistent &= !seastar::internal::case_insensitive_cmp()(resp->get_header("Connection"), "close");
```

Comment:
"&=" is not the right operator, it's the bitwise operator :-(
I know some people claim it works (https://stackoverflow.com/questions/2488406/why-doesnt-c-have-or-for-booleans) but I think it's wrong.

Reply:
By the way, "&=" also can't short-circuit - even if _persistent is false, it will still do (unnecessarily) the case_insensitive_cmp. I agree this has zero importance, but seeing "&=" still looks to me like a bug and any future reader will suspect it to be a bug, so let's not do it. Personally, I think the clearest thing to do is:
```
if (_presistent) {
    _persistent = !seastar::internal::case_insensitive_cmp()(resp->get_header("Connection"), "close");
}
```

Or if you want to support HTTP 1.0, assuming we have _http11 from parsing the HTTP response:
```
 // If the connection is already not to be persistent (e.g., handler failed), leave it like that.
// But if it could be persisted, the "Connection:" header in the server's response may force
// us to cancel the persistence, i.e., close the connection.
if (_presistent) {
    if (_http11) {
        _persistent = !seastar::internal::case_insensitive_cmp()(resp->get_header("Connection"), "close");
    } else {
         // HTTP 1.0 or 0.9
        _persistent = seastar::internal::case_insensitive_cmp()(resp->get_header("Connection"), "keep-alive");
     }
}
```

Reply:
This is true - in HTTP 1.0, if the client didn't ask for keep-alive, there is no way the server will set it.
There's always the risk that in the future we'll change the part of the code that sends the request, and forgot to change the part that parses the response, but I agree - all of this is NOT important and nobody will ever have an HTTP 1.0 server anyway.